### PR TITLE
add testing for agent status syncers.

### PR DESCRIPTION
**Pull request by morvencao**
ref: https://github.com/stolostron/backlog/issues/26232
Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

**Comment by morvencao** [comment]
Depends on https://github.com/stolostron/multicluster-global-hub/pull/270

File: agent/pkg/lease/lease.go [hunk 1]
```diff
@@ -35,9 +36,15 @@ type leaseUpdater struct {
 
 // AddHoHLeaseUpdater creates a new LeaseUpdater instance aand add it to given manager
 func AddHoHLeaseUpdater(mgr ctrl.Manager, addonNamespace, addonName string) error {
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		return err
+	var config *rest.Config
+	if isAgentTesting, ok := os.LookupEnv("AGENT_TESTING"); ok && isAgentTesting == "true" {
```
Comment (clyang82) [inline agent/pkg/lease/lease.go:40]: why we need this environment variable for testing?

Comment (morvencao) [reply agent/pkg/lease/lease.go:40]: just for skipping getting in-cluster kubeconfig for leader election, use kubeconfig of testEnv in testing.

**Pull request merged**